apps/src/applab/dataBrowser/ColumnHeader.jsx

```diff
@@ -0,0 +1,157 @@
+/**
+ * @overview Component for adding a new column to the specified table.
+ */
+
+import Radium from 'radium';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import * as dataStyles from './dataStyles';
+
+const styles = {
+  menu: {
+    float: 'right'
+  },
+  icon: {
+    color: 'white'
+  }
+};
+
+const ColumnHeader = React.createClass({
+  propTypes: {
+    columnName: React.PropTypes.string.isRequired,
+    columnNames: React.PropTypes.array.isRequired,
+    deleteColumn: React.PropTypes.func.isRequired,
+    editColumn: React.PropTypes.func.isRequired,
+    isEditable: React.PropTypes.bool.isRequired,
+    isEditing: React.PropTypes.bool.isRequired,
+    renameColumn: React.PropTypes.func.isRequired,
+  },
+
+  getInitialState() {
+    return {
+      newName: this.props.columnName,
```

Comment:
[Props in `getInitialState` is an antipattern.](https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html)